src/app/body/task-details/git/git.component.ts

```diff
@@ -18,59 +23,111 @@ export class GitComponent implements OnInit {
   @Input('taskId') taskId: string;
   @Input('orgDomain') orgDomain: string;
   prLink: string;
-  @Output() addedLink = new EventEmitter<{ completed: boolean, prLink: string }>();
+  @Output() addedPrLink = new EventEmitter<{ completed: boolean, prLink: string, prApiLink: string}>();
   componentName: string = "LINK"
   linkURL: string;
   linkType: string;
   enableLoader: boolean = false;
   showClose: boolean = false;
   prData: GitRepoData[] = [];
-
-  constructor(private httpService: HttpServiceService, private functions: AngularFireFunctions, public errorHandlerService: ErrorHandlerService, public validationService: ValidationService) { }
+  prNumber: number;
+  teamId:string;
+  team: Team
+  repoLink: string
+  noRepoLinked: boolean = false;
+  noPrExist: boolean = false;
+  prLinked: boolean = false;
+  prApiLink: string;
+  constructor(private httpService: HttpServiceService,public applicationSettingsService: ApplicationSettingsService, private startService: StartServiceService, private userService: UserServiceService, private backendService: BackendService, private functions: AngularFireFunctions, public errorHandlerService: ErrorHandlerService, public validationService: ValidationService) { }
 
   ngOnInit(): void {
-    this.getPullRequests();
-  }
-
-  async submit(url) {
-    this.prLink = url;
-    console.log(this.prLink)
-    this.enableLoader = true;
-    console.log(this.orgDomain, this.taskId)
-      const callable = this.functions.httpsCallable('linker/setLink');
-      this.linkType = "PR"
-      this.linkURL = this.prLink
-      console.log("this is the link url:", this.linkURL)
-      await callable({ OrgDomain: this.orgDomain, TaskID: this.taskId, LinkType: this.linkType, LinkURL: this.linkURL }).subscribe({
-        next: (data) => {
-          console.log("Successful ");
-          this.enableLoader = false;
-          this.showClose = true;
-          return;
-        },
-        error: (error) => {
-          this.enableLoader = false
-          this.errorHandlerService.showError = true;
-          this.errorHandlerService.getErrorCode(this.componentName, "InternalError", "Api");
-          console.log("Error", error);
-          console.error(error);
-        },
-        complete: () => console.info('Successful')
+    this.teamId = this.taskId.slice(0,3);
+    console.log(this.teamId);
+    if(this.startService.showTeams) {
+      this.getTeamDetails(this.teamId);
+    } else {
+      this.startService.userDataStateObservable.subscribe((data) => {
+        if(data){
+          this.getTeamDetails(this.teamId);
+        }
       });
-
+    }
+  }
+  getTeamDetails(teamId: string) {
+    this.applicationSettingsService.getTeamDetails(teamId).subscribe(data => {
+      this.team=data;
+      console.log(this.team.ProjectLink);
+      this.repoLink=this.team.ProjectLink;
+      if(this.repoLink!=""){
+      this.getPullRequests();
+      }
+      else{
+        this.noRepoLinked=true;
+       }
+    });
```

Comment:
This method contains the subscribe to trigger an api, just to get some data, I believe similar data can be sent from parent.

Reply:
now discussed, and no need to change anymore